### ISSUE TITLE
`prefer-const` false positive

### ISSUE THREAD

**Issue by SimenB** [eslint/eslint#5610]
**What version of ESLint are you using?**
2.4.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```json
{
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module",
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "rules": {
    "prefer-const": 2
  }
}
```

**What did you do? Please include the actual source code causing the issue.**
```jsx
import React from 'react';
import { browser } from '../../util/browserUtil';

let DevTools;

if (!browser.ie8()) DevTools = require('../devTools/devTools');

export default ({ children }) => (
  <div>
    <div>
      {children}
    </div>
    {DevTools && <DevTools />}
  </div>
);
```

**What did you expect to happen?**
No warning, as `DevTools` is used outside of the `if`-statement

**What actually happened? Please include the actual, raw output from ESLint.**
```
/home/testyo/repos/meeeeeeep/file.js
  6:5  error  'DevTools' is never reassigned, use 'const' instead  prefer-const
```

I'm aware of many potentially related issues (#5251, #5283, and #5284), but I don't think this one is covered

**Comment by btmills** [comment]
This is interesting. Tried these in the demo:

**Valid**:

```js
/* eslint prefer-const: 2 */
let foo;
if (bar) {
	foo = 42;
}
baz(foo);
```

**Invalid** (valid example with braces around `if` body removed):

```js
/* eslint prefer-const: 2 */
let foo;
if (bar) foo = 42;
baz(foo);
```

```
2:5 - 'foo' is never reassigned, use 'const' instead. (prefer-const)
```

**Invalid** (valid example with function call removed):

```js
/* eslint prefer-const: 2 */
let foo;
if (bar) {
	foo = 42;
}
```

```
2:5 - 'foo' is never reassigned, use 'const' instead. (prefer-const)
```

Why is it that the combination of curly braces around the `if` with a function call is considered valid, but if I remove either it errors?

@SimenB if you add curly braces around the `if` body, does it stop warning for you too?

**Comment by SimenB** [comment]
@btmills Ah, good catch! Adding braces makes the warning go away

**Comment by btmills** [comment]
Cool. So at least there's a workaround that can be used while we figure out what's going on here.

I don't see any [tests](https://github.com/eslint/eslint/blob/18559e217c05451779a1d554fe0bdfed8a987414/tests/lib/rules/prefer-const.js) that cover this case. @mysticatea you looked at scoping for `prefer-const` recently - any ideas as to what's going on here?

**Comment by mysticatea** [comment]
I has forgotten this case:

```js
/* eslint prefer-const: 2 */
let foo;
if (bar) foo = 42;
baz(foo);
```
